src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DatabaseHandler.java

```diff
@@ -0,0 +1,273 @@
+        String sqlUpdateStatement = String.format("UPDATE %s SET %s=? WHERE %s=?", table, EVENT_ID_KEY, table.keyName);
+        prepareAndExecuteStatement(sqlUpdateStatement, searchCriteria, eiffelEvent);
+    }
+
+    /**
+     * This function inserts values to the given table. The key value is
+     * different depending on Table (branch name for scs and change-id for scc)
+     *
+     * @param table
+     * @param key
+     * @param value
+     * @throws SQLException
+     * @throws ConnectException
+     */
+    public void insertInto(final Table table, final String key, final String value)
+            throws SQLException, ConnectException {
+        String sqlInsertStatement = String.format("INSERT INTO %s(%s,%s) VALUES(?,?)", table, EVENT_ID_KEY,
+                table.keyName);
+        prepareAndExecuteStatement(sqlInsertStatement, key, value);
+
+    }
+
+    /**
+     * This function will create a sql database structure in the database file if it
+     * does not exist.
+     *
+     * @throws ConnectException
+     */
+    private void createNewDatabase() throws ConnectException {
+        try (Connection sqlConnection = connect()) {
+            if (sqlConnection != null) {
+                String parentPath = buildParentFilePath();
+                createParentDirsIfNecessary(parentPath);
+
+                //When getMetaData is called, it will create the .db file
+                sqlConnection.getMetaData();
+                sqlConnection.close();
+            }
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to create new database: {}\n{}", e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This function loops the values in the Table enum and executes the create
+     * table command for each table and creates it in the database if it does not already
+     * exist.
+     *
+     * @throws ConnectException
+     */
+    private void createTables() throws ConnectException {
+        try (Connection connection = connect(); Statement statement = connection.createStatement()) {
+            for (Table table : Table.values()) {
+                createTable(table, statement);
+            }
+            LOGGER.debug("Created tables successfully");
+        } catch (SQLException e) {
+            LOGGER.error("Error while creating Tables in database: {}\n{}", e.getMessage(), e);
+        }
+    }
+
+    /**
+     * This function connects to the database and returns a connection
+     *
+     * @return Connection to database
+     * @throws ConnectException
+     */
+    private Connection connect() throws ConnectException {
+        try {
+            Class.forName("org.sqlite.JDBC");
+            return DriverManager.getConnection(this.databaseFile);
+        } catch (SQLException | ClassNotFoundException e) {
+            throw new ConnectException(
+                    String.format("Failed to create a database connection. %s\n%s", e.getMessage(), e));
```

Comment:
This I also believe to be wrong, Exceptions usually support exception as a parameter. This could then be written as: throw new ConnectException("Failed to create a database connection.", e);